code/components/citizen-server-impl/src/state/ServerGameState.cpp

```diff
@@ -2735,10 +2735,11 @@ bool ServerGameState::SetEntityStateBag(uint8_t playerId, uint16_t objectId, std
 {
 	if (auto entity = GetEntity(0, objectId))
 	{
-		if (entity->GetStateBag())
+		if (entity->HasStateBag())
```

Comment:
Maybe add a method that uses compare_exchange internally to set the boolean to true if the expected state is false. This way the atomic is using just one operation and it seems to fit both into this code and into the ENSURE_ENTITY_STATE_BAG logic as well.